src/Microsoft.Diagnostics.Monitoring.EventPipe/Configuration/MetricSourceConfiguration.cs

```diff
@@ -65,7 +68,10 @@ public MetricSourceConfiguration(float metricIntervalSeconds, IEnumerable<Metric
                 const long TimeSeriesValuesEventKeyword = 0x2;
                 string metrics = string.Join(',', meterProviders.Select(p => p.Provider));
 
-                SessionId = Guid.NewGuid().ToString();
+                ClientId = Guid.NewGuid().ToString();
+
+                // Shared Session Id was added in 8.0 - older runtimes will not properly support it.
+                SessionId = (version != null && version.Major >= 8) ? SharedSessionId : Guid.NewGuid().ToString();
```

Comment:
This is assuming the runtime is coreclr; is there an affordance for mono? Not sure if there is a way to detect that from the perspective of an out-of-proc diagnostic tool.

Reply:
@noahfalk Would you be able to weigh in on this or point me in the right direction to investigate this further?

Reply:
Mono uses the same System.Private.CoreLib as coreclr, unless we specifically did something to exclude mono from using the metrics stuff I would expect it to be on mono too

Reply:
Is `version` refering to the assembly version of System.Diagnostics.DiagnosticSource.dll? Assuming it is that then there should be no difference between Mono, CLR, or any other runtime because they are all sharing the same S.D.DiagnosticSource assembly that has the same version.

Reply:
> Is `version` refering to the assembly version of System.Diagnostics.DiagnosticSource.dll? Assuming it is that then there should be no difference between Mono, CLR, or any other runtime because they are all sharing the same S.D.DiagnosticSource assembly that has the same version.

From an out-or-proc perspective, we can't get the assembly version, especially not in multi-container environments. This version seems to be the runtime version, which would be 8.0.0 for CoreCLR but that may not be true for Mono.

Reply:
> This version seems to be the runtime version

Where did the version come from? An API call, a file header, a file path, something sent across an IPC channel? 

If it is the runtime version it would be a little unfortunate that you won't support the scenario where a .NET 6 app uses a Nuget package reference for DiagnosticSource version 8.0, but its certainly not a requirement that you support it.

Reply:
> > This version seems to be the runtime version
> 
> Where did the version come from? An API call, a file header, a file path, something sent across an IPC channel?
> 
> If it is the runtime version it would be a little unfortunate that you won't support the scenario where a .NET 6 app uses a Nuget package reference for DiagnosticSource version 8.0, but its certainly not a requirement that you support it.

As of right now, it's coming from the call to DiagnosticsClient.GetProcessInfo call. I haven't thought about a scenario involving an app developer using a newer version of that package in their lower TFM apps. That might be tough to do  given that .NET Monitor is typically does not share a process namespace with the target app so we can't gleam what version that package would be, especially during the diagnostic startup suspension.

Maybe it's better to make this a `bool useSharedSession` parameter and let the caller determine if shared sessions should be used given the capabilities of the caller.

Reply:
> Maybe it's better to make this a bool useSharedSession parameter and let the caller determine if shared sessions should be used given the capabilities of the caller.

For dotnet-monitor does that mean the same version check is just happening in a different place in the code, or it actually changes how we'd do the check? I was more looking at would the check produce a useful result rather than where the check occurs. If dotnet-monitor doesn't get the new behavior for .NET 6.0 apps running 8.0 DiagnosticSource I'm not going to lose sleep over it.

@davmason - based on the docs at https://github.com/dotnet/diagnostics/blob/main/documentation/design-docs/ipc-protocol.md#processinfo2 the text isn't super descriptive, but it implies that Mono would return 8.0.*. Do you know if it does or can you find out? On the one hand I suspect there is very little usage of dotnet-monitor with Mono so it probably doesn't matter if this didn't work, on the other I feel like we should clearly document the behavior of that version string regardless of its role in this feature.

Reply:
> For dotnet-monitor does that mean the same version check is just happening in a different place in the code, or it actually changes how we'd do the check? I was more looking at would the check produce a useful result rather than where the check occurs. If dotnet-monitor doesn't get the new behavior for .NET 6.0 apps running 8.0 DiagnosticSource I'm not going to lose sleep over it.

I'm not sure we have much of an option here for .NET Monitor. We would need some way of interrogating the app folder in the file system in order to find that particular assembly and get its version. And consider that against the runtime version of in is less than 8. This would require a new diagnostic command that would get us the assembly information from the app folder.

Reply:
Yeah, I'm not aware you have a good alternative. Long term we could of course add one, but also long term it won't matter because all the builds that had an assembly specific version check would also have DiagnosticSource 8.0.

Reply:
@jander-msft I made the change to remove the version logic from `MetricSourceConfiguration` - let me know if that's what you wanted.

It sounds like on the `dotnet-monitor` side of things we've settled on checking that the runtime version >= 8, but the `dotnet counters` logic is still up in the air. If there's concerns about the granularity/correctness of the current logic that will require more investigation, Justin and I were chatting yesterday about temporarily disabling the SHARED session in dotnet-counters as part of this PR. If we go that route, I could either pull all the new logic out of `CounterMonitor` to revert it to its existing state, or I could leave it in and we never toggle the SHARED switch (this would be leaving unused code behind, but if the expectation is that this would be enabled in the near future, everything would already be rigged up correctly). I've left it as-is for now, but just let me know if there's any action you want me to take and I'll push another update.

Reply:
I'm happy to have @davmason make the call.

Reply:
For the mono question, from my research it looks like the mono version will be 8.0 (match the coreclr version).

For the dotnet-counters logic, I think matching dotnet-monitor is fine and we don't need special logic.